src/util/cdn.py

```diff
@@ -0,0 +1,38 @@
+"""Helper functions for downloading files from CDN links."""
+
+import os
+from pathlib import Path
+from urllib.parse import urlparse
+
+import requests
+
+
+def _get_filename(url: str) -> str:
+    """Extract file name from last part of CDN link.
+
+    Args:
+        url: The CDN link to extract file name from
+
+    Returns:
+        The last part of the CDN link, to be used as the file name, or an empty 
+        string if the URL is invalid or has no path.
+    """
+    parsed_url = urlparse(url)
+    if not parsed_url.path:
+        return ""
+    return os.path.basename(parsed_url.path)
```

Comment:
This is not a reliable thing to do. Use the filename from the curseforge API query to name the file.

Reply:
Fair enough, this particular CurseForge API call unfortunately only returns the CDN link. We can just require the caller of this function to specify the filename instead, and grab the filename from the lockfile entry itself.